components/screens/StartScreen.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { View, StyleSheet, Text, ImageBackground, Dimensions, TouchableOpacity } from 'react-native';
+
+import Carousel, { Pagination } from 'react-native-snap-carousel';
+import LinearGradient from 'react-native-linear-gradient';
+
+import { useSelector, useDispatch } from 'react-redux';
+import { setIsSkipped } from "../../store/guideReucer"
+import { useNavigation } from "@react-navigation/native";
+
+function StartScreen(props) {
+    const data = useSelector((state) => state.getCarouselData.data);
+    const { isSkipped } = useSelector((state) => state.isGuideSkipped)
+    const [index, setIndex] = useState(0);
+
+    const carouselRef = useRef();
+    const navigation = useNavigation();
+    const dispatch = useDispatch();
+
+
+    if (isSkipped) {
+        navigation.navigate("LoginScreen");
+    } else {
+        navigation.navigate("StartScreen");
+    }
+
+
+    const goNextPage = () => {
+        setIndex(index + 1);
+        carouselRef.current.snapToNext(index);
+    }
+
+    const handleGetStartedPress = () => {
+        dispatch(setIsSkipped())
+    }
+
+    const skipGuide = () => {
+        dispatch(setIsSkipped())
+    }
+
+    function renderItem({ item }) {
+        return (
+            <ImageBackground
+                source={item.image}
+                style={styles.carouselBackgroundContainer}
+
+            >
+                <LinearGradient
+                    colors={[
+                        "rgba(255, 255, 255, 0)",
+                        "rgba(255, 255, 255, 1)"
+                    ]}
+                    locations={[0, 0.7]}
+                    style={{ flex: 1 }}
```

Comment:
the inline style is a bad performance issue, react native will create a new object each time the component is rerendered please always use StyleSheet.create for writing static style